content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,148 @@
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary privileges.
+
+## Extensions prerequisites
+
+Prisma Client extensions are available in Preview from version [4.7.0](https://github.com/prisma/prisma/releases/tag/4.7.0).
+
+To use Prisma Client extensions, you must enable the `clientExtensions` preview feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create an extension
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an extended client. In your `$extends` structure, use the appropriate extension component or components (`model`, `client`, `result` or `query`).
+
+For example, the following code uses a [`model` extension](/concepts/components/prisma-client/client-extensions/model) to extend the `User` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name: `signUp`, // Optional: name appears in error logs
+  model: {        // This is a `model` component
+    user: { ... } // The extension logic for the `user` model goes inside the curly braces
+  },
+})
+```
+
+## Name an extension for error logs
+
+You can name your extensions to help identify them in error logs. To do so, use the optional field `name`. For example:
+
+```ts
+const xprisma = prisma.$extends({
+  name: `signUp`,  // (Optional) Extension name
+  model: {
+    user: { ... }
+ },
+})
+```
+
+## Multiple extensions
+
+You can associate an extension with an [extended client](#about-prisma-client-extensions) in one of two ways:
+
+- You can associate it with an extended client on its own, or
+- You can combine the extension with other extensions and associate all of these extensions with an extended client. The functionality from these combined extensions applies to the same extended client.
+  Note: [Combined extensions can conflict](#conflicts-in-combined-extensions).
+
+You can combine the two approaches above. For example, you might associate one extension with its own extended client and associate two other extensions with another extended client. [Learn more about how client instances interact](#extended-clients).
+
+### Apply multiple extensions to an extended client
+
+In the following example, suppose that you have two extensions, `extensionA` and `extensionB`. There are two ways to combine these.
+
+#### Option 1: Declare the new client in one line
+
+With this option, you apply both extensions to a new client in one line of code.
+
+```ts
+// First of all, store your original Prisma Client in a variable as usual
+const prisma = new PrismaClient()
+
+// Declare an extended client that has an extensionA and extensionB
+const prismaAB = prisma.$extends(extensionA).$extends(extensionB)
+```
+
+You can then refer to `prismaAB` in your code, for example `prismaAB.getName()`.
```

Comment:
GetName feels confusing

Reply:
Maybe `myExtensionMethod()` is more clear

Reply:
Good call, I'll change it